File: examples-sdk/python/014-basic-kafka-stream.py
#!/usr/bin/env python3
"""014-basic-kafka-stream — Python SDK equivalent
Compile: vil compile --from python --input 014-basic-kafka-stream.py --release
"""
import os
from vil import VilPipeline, VilServer, ServiceProcess

server = VilServer("kafka-stream", port=8080)
kafka = server.service_process("kafka")
kafka.endpoint("GET", "/kafka/config", "kafka_config")
kafka.endpoint("POST", "/kafka/produce", "kafka_produce")
kafka.endpoint("GET", "/kafka/consumer", "consumer_info")
kafka.endpoint("GET", "/kafka/bridge", "bridge_status")
root = server.service_process("root")
server.compile()

File: examples-sdk/typescript/014-basic-kafka-stream.ts
#!/usr/bin/env tsx
// 014-basic-kafka-stream — TypeScript SDK equivalent
// Compile: vil compile --from typescript --input 014-basic-kafka-stream.ts --release

import { VilPipeline, VilServer, ServiceProcess } from "vil-sdk";

const server = new VilServer("kafka-stream", 8080);
const kafka = new ServiceProcess("kafka");
kafka.endpoint("GET", "/kafka/config", "kafka_config");
kafka.endpoint("POST", "/kafka/produce", "kafka_produce");
kafka.endpoint("GET", "/kafka/consumer", "consumer_info");
kafka.endpoint("GET", "/kafka/bridge", "bridge_status");
server.service(kafka);
const root = new ServiceProcess("root");
server.service(root);
server.compile();
